### PR TITLE
Get rid of InspectableQueue in AvroSpanExporter

### LINKED ISSUE #340: AvroSpanExporter shouldn't check the queue for elements on release
TODO:
1. Rely on the semaphore only.
2. Fix the check for acquired permits.
3. Add an optional parameter for the number of worker fibers (under discussion).

### PR DESCRIPTION
**Pull request by catostrophe**
Closes #340

### CODE REVIEW

File: modules/avro-exporter/src/main/scala/io/janstenpickle/trace4cats/avro/AvroSpanExporter.scala [hunk 1]
```diff
@@ -69,40 +75,39 @@ object AvroSpanExporter {
...
           .compile
           .drain
           .start
       )(fiber =>
-        Applicative[F].unit.whileM_(for {
-          queueNonEmpty <- queue.getSize.map(_ != 0)
-          semaphoreSet <- semaphore.count.map(_ == 0)
-          _ <- Timer[F].sleep(50.millis)
-        } yield queueNonEmpty || semaphoreSet) >> fiber.cancel
+        Timer[F].sleep(50.millis).whileM_(semaphore.available.map(_ < maxPermits)) >>
```
Comment (catostrophe) [inline modules/avro-exporter/src/main/scala/io/janstenpickle/trace4cats/avro/AvroSpanExporter.scala:107]: This is probably not the best strategy for releasing multiple fibers, since every single fiber waits until no work is being done at all.

File: modules/avro-exporter/src/main/scala/io/janstenpickle/trace4cats/avro/AvroSpanExporter.scala [hunk 2]
```diff
@@ -112,7 +117,11 @@ object AvroSpanExporter {
     blocker: Blocker,
     host: String = agentHostname,
     port: Int = agentPort,
+    numFibers: Int = 1
```
Comment (janstenpickle) [inline modules/avro-exporter/src/main/scala/io/janstenpickle/trace4cats/avro/AvroSpanExporter.scala:120]: Could you modify the tests to add a case where multiple fibers are used please?

Comment (catostrophe) [reply modules/avro-exporter/src/main/scala/io/janstenpickle/trace4cats/avro/AvroSpanExporter.scala:120]: Yes, a bit later. How important is the order of spans being sent by Exporter? Multiple fibers will be sending batches out of the queue order.

Comment (catostrophe) [reply modules/avro-exporter/src/main/scala/io/janstenpickle/trace4cats/avro/AvroSpanExporter.scala:120]: @janstenpickle done
I also noticed that the existing tests were using `Batch[List]` which arb instance returned zero-sized lists (there's minSuccessful = 1 configuration). I changed it to `Batch[Chunk]` as it's used in other tests

File: modules/avro-exporter/src/main/scala/io/janstenpickle/trace4cats/avro/AvroSpanExporter.scala [hunk 3]
```diff
@@ -59,7 +62,10 @@ object AvroSpanExporter {
...
```
Comment (catostrophe) [inline modules/avro-exporter/src/main/scala/io/janstenpickle/trace4cats/avro/AvroSpanExporter.scala:65]: Add documentation for `numFibers` with warning

**Review by janstenpickle — APPROVED** [review]
Pending scaladoc, this is good!
